nft_ingester/src/program_transformers/bubblegum/creator_verification.rs

```diff
@@ -82,15 +85,18 @@ where
 
                 // Partial update of asset table with just leaf owner and delegate.
                 upsert_asset_with_owner_and_delegate_info(
-                    txn,
+                    &multi_txn,
                     id_bytes.to_vec(),
                     owner_bytes,
                     delegate,
                     seq as i64,
                 )
                 .await?;
 
-                upsert_asset_with_seq(txn, id_bytes.to_vec(), seq as i64).await?;
+                upsert_asset_with_seq(&multi_txn, id_bytes.to_vec(), seq as i64).await?;
+
+                // Close out transaction and relinqish the lock.
+                multi_txn.commit().await?;
```

Comment:
You need to know how to handle errors here. what if the transaction fails? Need to have recovery scenarios in that acse surely?

Reply:
So the way it works with SeaORM, if the the transaction (`multi_txn` in this case) goes out of scope, it is rolled back.  Therefore, the way the code is structured, if there is an error on one of the executions of the multiple statement transaction, there will be an early return, `multi_txn` will go out of scope and the transaction will be rolled back.

The reported error will be the one that caused the early return, which is how it functions today.  So I think the only difference will be that the `asset` table changes are tied together, and in the case of failure, all of them will be rolled back together.  I don't know if this could be considered better behavior than today or not; it seems about equivalent to me, because either way some of the asset updates for an instruction were missed, and the asset state overall would be missing some updates.